lib/python/rose/config.py

```diff
@@ -1580,25 +1600,26 @@ def load(source, root=None):
 
 def sort_element(elem_1, elem_2):
     """Sort pieces of text, numerically if possible."""
-    if elem_1.isdigit():
-        if elem_2.isdigit():
-            return cmp(int(elem_1), int(elem_2))
+    if elem_1.isdigit() and elem_2.isdigit():
+        return (int(elem_1) > int(elem_2)) - (int(elem_1) < int(elem_2))
```

Comment:
Probably my 👓 , but are we applying a subtraction to two booleans here?

Reply:
We are indeed. This is a snipped designed to have the same effect as `cmp`. Going away and thinking about this!